crates/wasi-common/src/sys/unix/hostcalls_impl/fs.rs

```diff
@@ -60,16 +61,9 @@ pub(crate) fn fd_advise(
     unsafe { posix_fadvise(file.as_raw_fd(), offset, len, host_advice) }.map_err(Into::into)
 }
 
-pub(crate) fn path_create_directory(resolved: PathGet) -> Result<()> {
+pub(crate) fn path_create_directory<'a>(base: &File, path: &str) -> Result<()> {
```

Comment:
```suggestion
pub(crate) fn path_create_directory(base: &File, path: &str) -> Result<()> {
```

Looks like an explicit lifetime is not needed here after all.